daemons/fenced/fenced_commands.c

```diff
@@ -708,10 +706,10 @@ get_agent_metadata(const char *agent)
 
     init_metadata_cache();
     buffer = g_hash_table_lookup(metadata_cache, agent);
-    if(pcmk__str_eq(agent, STONITH_WATCHDOG_AGENT, pcmk__str_casei)) {
+    if(pcmk__str_eq(agent, STONITH_WATCHDOG_AGENT_INTERNAL, pcmk__str_none)) {
         return NULL;
```

Comment:
Bah, that means we need a fix for this particular issue in 2.1.1. I suppose for that release we could just skip the retry timer if the id is AGENT. Or if you're really confident in this, we could make it a feature release ...

Reply:
Think we need to do the mapping just once in get_agent_metadata if we use the mapped name just for the api-call and not for the cache. Don't think we would get issues with the "name" attribute being actually wrong.

We could as well switch the get_agent_metadata prototype to rv + pointer and signal via rv if it makes sense to do a retry. So the special handling for AGENT_INTERNAL would stay in a single place.

Reply:
Pacemaker uses the meta-data to see what actions are supported (without metadata, we will assume that list, status, reboot, and on are not supported), and to see whether port or plug is supported (without metadata, we assume not).

Coincidentally those assumptions will not cause any problems for watchdog fencing which is why this hasn't come up before. However it does mean our support for "list" is ignored. We don't look for or display the agent name from the meta-data, so it wouldn't matter if we reused the fence_watchdog metadata for #watchdog.

Since we now have metadata (via fence_watchdog), I lean to taking advantage of it. Alternatively we could define a C string constant with minimal metadata for #watchdog. But leaving it NULL and just letting the assumptions happen is fine (especially for 2.1.1).

A standard return code for get_agent_metadata() is reasonable, we can use EAGAIN to indicate "try again later".